### PR TITLE
Refactor image-size test to follow Playwright's best practices

### PR DESCRIPTION
**Pull request by kevin940726**
## What?

Continued from #40467. Refactor `image-size.spec.js` to follow the [best practices](https://github.com/WordPress/gutenberg/blob/HEAD/test/e2e/README.md#best-practices).

This PR also fixes an ESLint issue that is discovered when creating this PR.

## Why?

While #40467 did a great job migrating the test to Playwright, there were some missing pieces in the playwright utils preventing it from adopting the latest best practices. This PR tries to fill in the gap and also refactor the test to follow the best practices in the process.

## How?

Introduced some `media` request utils and use role selectors to query elements.

## Testing Instructions

```sh
npm run test-e2e:playwright -- test/e2e/specs/editor/plugins/image-size.spec.js
```

CI should pass.

### CODE REVIEW

File: test/e2e/specs/editor/plugins/image-size.spec.js
Comment (kevin940726) [inline test/e2e/specs/editor/plugins/image-size.spec.js:31]: This also normalize the path so that it can be run on Windows.

File: test/e2e/specs/editor/plugins/image-size.spec.js
Comment (kevin940726) [inline test/e2e/specs/editor/plugins/image-size.spec.js:28]: We already have dedicated e2e tests to test the uploading spec in `editor/blocks/image.test.js` so we don't have to use the UI to upload the file here.

This got me thinking though, why didn't we write this test in `editor/blocks/image.test.js`? Is there a reason why we created this test file separately?

File: test/e2e/specs/editor/plugins/image-size.spec.js
Comment (kevin940726) [inline test/e2e/specs/editor/plugins/image-size.spec.js:57]: TBH, this isn't really specific but seems to be the best we can do now. I wonder if this is actually a bug accessibility-wise? For instance, maybe it should be under an "Image size" section, rather than relying on the order. But I guess reading orders are also accessible in some way 🤔 .

File: packages/e2e-test-utils-playwright/src/request/index.ts
Comment (kevin940726) [inline packages/e2e-test-utils-playwright/src/request/index.ts:126]: Off-topic: I'm starting to think that maybe we should namespace these by file names. So `requestUtils.listMedia` becomes `requestUtils.media.list`.

Comment (glendaviesnz) [reply packages/e2e-test-utils-playwright/src/request/index.ts:126]: How are you thinking that would look in terms of the exports, imports?

**Comment by glendaviesnz** [comment]
This tested well for me on multiple runs. Did you want to wait and see if anybody has any thoughts on why this was set up separate to the other image tests before we merge this? Happy give a signoff on it if not.

**Comment by kevin940726** [comment]
Did you want to wait and see if anybody has any thoughts on why this was set up separate to the other image tests before we merge this?

We can discuss them in another issue or PR 👍, just wanted to bring them up with context so that I wouldn't forget 😆.

File: packages/e2e-test-utils-playwright/src/request/index.ts
Comment (kevin940726) [reply packages/e2e-test-utils-playwright/src/request/index.ts:126]: I don't think that would be much different? Either way it should be implementation detail and we usually don't care about tree-shaking for e2e test utils anyway.

I'm thinking maybe `media.js` can export a class and `requestUtils` can then initiate an instance.

```js
// request/media.js
export class MediaRequestUtils { /* ... */ };

// request/index.ts
media = new MediaRequestUtils( this );
```

Comment (glendaviesnz) [reply packages/e2e-test-utils-playwright/src/request/index.ts:126]: It seems like Classes can end up as superfluous wrappers in js modules, unless there is a reason to want to maintain state across methods in an OOP style way. Just making the [existing object export ](https://github.com/WordPress/gutenberg/pull/40644/files#diff-3cbf286b73d9eb85870b166c2f82e0efca40ae21ddede2be27082b82e8d83eaeR102) a default export and importing as `import media from from './media'` would be another approach and no newing up required ... but I don't have a strong opinion either way.

File: test/e2e/specs/editor/plugins/image-size.spec.js
Comment (glendaviesnz) [reply test/e2e/specs/editor/plugins/image-size.spec.js:57]: I couldn't see a more accurate way of handling this currently either

**Review by glendaviesnz — APPROVED** [review]
LGTM, passes locally on multiple runs, and in CI.

**Pull request merged**
